### PR TITLE
updated role to use latest ansible; template ldap and registry

### PR DESCRIPTION
**Pull request by asaphe**
Please review the changes to the role
- Updated to latest ansible modules
- Add pip docker-py as a prerequisite
- Add the ability to define LDAP and other settings via a jinja2 template
- Add the ability to define a registry via a jinja2 template
- Add the ability to define multiple endpoints using ansible dict

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -1,9 +1,43 @@
...
-portainer_version: 1.11.0
-portainer_port: 5000
-portainer_admin_pass: admin555
-
-swarm_nodes_group: swarm-nodes
-swarm_managers_group: swarm-managers
-
-mode: cluster
+### Portainer version
+version: develop
```
Comment (hagzag) [inline defaults/main.yml]: I think the default should use a stable release or if we remain with develop branch we should explain why in the README

File: defaults/main.yml [hunk 2]
```diff
@@ -1,9 +1,43 @@
...
+ldap_url: 'ldap.mycompany.com'
+ldap_port: 636
+tls_enabled: true
+tls_skipverify: true
+start_tls: true
+ldap_base_dn: 'ou=Users,o=org,dc=mycompany,dc=com'
+ldap_filter: '(objectClass=inetOrgPerson)'
+ldap_username_attribute: 'uid'
+### Registry
+docker_registery: true
```
Comment (hagzag) [inline defaults/main.yml]: I think this should be false by default ... and perhaps the entire nexus section optional

File: tasks/main.yml [hunk 3]
```diff
@@ -1,52 +1,93 @@
...
+    url: "{{ portainer_endpoint }}/registries"
     method: POST
     return_content: yes
     headers:
       Authorization: "{{ (auth_token.content|from_json).jwt }}"
     body_format: json
-    body: {"Name":"{{ remote_endpoint_name }}", "URL":"tcp://{{ remote_endpoint_ip }}:{{ remote_endpoint_port }}"}
-  when: remote_endpoint is defined
+    body: "{{ lookup('template','registry.json.j2') }}"
+  when: docker_registery
```
Comment (hagzag) [inline tasks/main.yml]: Again - this should be `false` by default

File: defaults/main.yml (hunk 2 above)
Comment (asaphe) [reply defaults/main.yml]: fixed

File: defaults/main.yml (hunk 1 above)
Comment (asaphe) [reply defaults/main.yml]: fixed by explaining it in the README and changing to latest

File: tasks/main.yml (hunk 3 above)
Comment (asaphe) [reply tasks/main.yml]: fixed

**Comment by asaphe** [comment]
Ready for review

**Pull request merged**